examples/static-fileserver/README.md

```diff
@@ -0,0 +1,25 @@
+# Spin static fileserver example
```

Comment:
It relates in that I use it in my unit testing. I needed a sample Spin app that used static files. Kind of a kludge. If you don't want this example here then where should I put sample apps that my unit tests depend on?

```rust
    #[tokio::test]
    async fn test_asset_path_patterns() {
        let app_path = "examples/static-fileserver/spin.toml";
        let watch_command = WatchCommand {
            app: app_path.into(),
            clear: false,
            debounce: 100,
            skip_build: false,
            skip_up: false,
            watch_wasm: true,
            up_args: vec!["--quiet".into()],
        };
        let path_patterns = watch_command.generate_path_patterns().await.unwrap();
        assert_eq!(path_patterns.get(0), Some(&String::from("assets/**/*")));
    }
```

Reply:
Ah, okay.  I think we've tried to put test apps in `tests` (either top level or in the crate that the test applies to), because examples are aimed at readers and therefore might be changed to demo new features or to better communicate - an 'example' doesn't connote that something might have a dependency on it.  (We probably do have older tests that use examples, but I believe we are trying to move away from that to dedicated test cases.)

Reply:
My concern with putting it under `tests` is that people might think that `integration.rs` or `spinup_tests.rs` depend on it when really it's the unit tests in `watch.rs` that depend on it (it being the example spin app). Perhaps that is better than depending on the `examples` folder though.

If you don't respond here I'll take silence to mean that you still think I should put it in `tests` despite that potential extra confusion.

Reply:
You could put it with the test where it's used.  Cf. https://github.com/fermyon/spin/tree/main/crates/loader/tests

Reply:
Apologies for dragging this on but I feel like that also doesn't make sense b/c the file is in the main src directory `src/commands/watch.rs`. Adding a `src/commands/tests` feels off to me but what do you think?

Reply:
I have no particular problem with `/src/tests` but if you feel `tests` should be adjacent to `src` not under it then that kind of implies an answer _grin_.  You could make a `/tests/watch` directory (next to `/tests/build`) with a read-me to explain what it relates to.  I'm kind of not fussed, I just don't want us taking dependencies on `examples`.

Reply:
Cool! I'll pick one of those that I think makes sense